src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/TriggerHelperBindingProvider.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    // Regular BindToInput has to do a TAttribute --> Value creation. 
+    // But triggers already have a Listener that provided the initial object; and we're just
+    // converting it ot the user's target parameter. 
+    internal class TriggerHelperBindingProvider<TAttribute, TTriggerValue> :
```

Comment:
wonder if this shouldn't be renamed to TriggerAdapterBindingProvider?

Reply:
will do. 

---
In reply to: [150925974](https://github.com/Azure/azure-webjobs-sdk/pull/1422#discussion_r150925974) [](ancestors = 150925974)